### DISCUSSION TITLE
How to use smart form methods in parent component?

### DISCUSSION THREAD

**Discussion by abizden** [react-hook-form/react-hook-form#9805] [General]
Smart How can I pass methods in form to a parent component. I'm sending it with the props method, but this time the formstate issubmitting doesn't work quite right. 

While using the <Form component as in the image, I defined a method called onLoad. I send hook form methods as parameters to this method during the useEffect component loading phase from within the form component itself. But when I submit the form afterwards, the isSubmitting value or other values ​​still come as the old value. When I remove Smartform and use it with normal basic form flow, there is no problem. How do you think I can solve this with the best method?

The page where I'm using the form component
![Screen Shot 2023-01-19 at 14 39 57](https://user-images.githubusercontent.com/13751058/213433169-7ea97502-8013-49d1-a03d-b393985769ad.png)

Smart Form Component
![Screen Shot 2023-01-19 at 14 40 25](https://user-images.githubusercontent.com/13751058/213433263-dffeb91a-0499-4a77-bc5a-d98b2e00a232.png)

**Comment by bluebill1049** [comment]
It's a design choice, perhaps you want to move hook out from the Form component and then you will have access at the parent level and pass all methods to the Form as props.

**Reply by antonsamper** [reply]
This should really be marked as the solution! I’ve locally implemented passing form methods from the parent to the smart <Form /> component, and it works perfectly. Definitely the simplest approach I’ve come across.
